tap_github/repository_streams.py

```diff
@@ -2625,3 +2625,131 @@ class TagsStream(GitHubRestStream):
+        # Deployment Keys
+        th.Property("url", th.StringType),
+        th.Property("id", th.IntegerType),
+        th.Property("node_id", th.StringType),
+        th.Property("sha", th.StringType),
+        th.Property("ref", th.StringType),
+        th.Property("task", th.StringType),
+        th.Property("payload", th.StringType),
+        th.Property("original_environment", th.StringType),
+        th.Property("environment", th.StringType),
+        th.Property("description", th.StringType),
+        th.Property(
+            "creator",
+            th.ObjectType(
+                th.Property("login", th.StringType),
+                th.Property("id", th.IntegerType),
+                th.Property("node_id", th.StringType),
+                th.Property("avatar_url", th.StringType),
+                th.Property("gravatar_id", th.StringType),
+                th.Property("url", th.StringType),
+                th.Property("html_url", th.StringType),
+                th.Property("followers_url", th.StringType),
+                th.Property("following_url", th.StringType),
+                th.Property("gists_url", th.StringType),
+                th.Property("starred_url", th.StringType),
+                th.Property("subscriptions_url", th.StringType),
+                th.Property("organizations_url", th.StringType),
+                th.Property("repos_url", th.StringType),
+                th.Property("events_url", th.StringType),
+                th.Property("received_events_url", th.StringType),
+                th.Property("type", th.StringType),
+                th.Property("site_admin", th.BooleanType),
+            ),
+        ),
+        th.Property("created_at", th.DateTimeType),
+        th.Property("updated_at", th.DateTimeType),
+        th.Property("statuses_url", th.StringType),
+        th.Property("repository_url", th.StringType),
+        th.Property("transient_environment", th.BooleanType),
+        th.Property("production_environment", th.BooleanType),
+    ).to_dict()
+
+    def get_child_context(self, record: dict, context: dict | None) -> dict:
+        """Return a child context object from the record and optional provided context.
+        By default, will return context if provided and otherwise the record dict.
+        Developers may override this behavior to send specific information to child
+        streams for context.
+        """
+        return {
+            "org": context["org"] if context else None,
+            "repo": context["repo"] if context else None,
+            "deployment_id": record["id"],
+            "repo_id": context["repo_id"] if context else None,
+        }
+
+
+class DeploymentStatusesStream(GitHubRestStream):
+    """A stream dedicated to fetching deployment statuses in a repository."""
+
+    name = "deployment_statuses"
+    path = "/repos/{org}/{repo}/deployments/{deployment_id}/statuses"
+    primary_keys: ClassVar[list[str]] = ["repo", "org", "deployment_id"]
```

Comment:
Similarly, I think we could away with

```suggestion
    primary_keys: ClassVar[list[str]] = ["node_id"]
```